lexicon/providers/valuedomain.py

```diff
@@ -0,0 +1,227 @@
+"""Module provider for Value Domain"""
+from __future__ import absolute_import
+
+import hashlib
+import json
+import logging
+import re
+import requests
+from urllib.parse import urlencode
+from lexicon.providers.base import Provider as BaseProvider
+
+LOGGER = logging.getLogger(__name__)
+
+NAMESERVER_DOMAINS = ["value-domain.com", "dnsv.jp"]
+
+
+def provider_parser(subparser):
+    """Generate a provider parser for Value Domain"""
+    subparser.add_argument(
+        "--auth-token", help="specify access token for authentication"
+    )
+
+
+class Provider(BaseProvider):
+    """Provider class for Value Domain"""
+
+    def __init__(self, config):
+        super(Provider, self).__init__(config)
+        self.nameserver = "valuedomain1"
+        self.ttl = 3600
+        self.domain_id = None
+        self.api_endpoint = "https://api.value-domain.com/v1"
+
+    def _authenticate(self):
+        payload = self._get("/domains/{0}/dns".format(self.domain))
+        self.ttl = int(payload["results"]["ttl"])
+        self.nameserver = payload["results"]["ns_type"]
+
+        if payload["results"]["domainname"] == self.domain:
+            self.domain_id = payload["results"]["domainid"]
+            return
+
+        raise Exception("No domain found")
+
+    # Create record. If record already exists with the same content, do nothing'
+    def _create_record(self, rtype, name, content):
+        name = self._relative_name(name)
+        resource_record_sets = self._get_resource_record_sets()
+        index = self._find_resource_record_set(
+            resource_record_sets, identifier=None, rtype=rtype, name=name, content=content
+        )
+        if index >= 0:
+            LOGGER.debug("record already exists")
+            return True
+
+        resource_record_sets.append((rtype.lower(), name, self._bind_format_target(rtype, content)))
+        self._update_resource_record_sets(resource_record_sets)
+        LOGGER.debug("create_record: %s", True)
+        return True
+
+    @staticmethod
+    def _identifier(record):
```

Comment:
Note to myself. I really need to make this function generally available, since it is used in almost all providers that do not have ID native support.